lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -172,13 +172,12 @@ def run!(verify_result: true, allow_protected_ns: false, prune: true)
       end
       ::StatsD.measure('all_resources.duration', StatsD.duration(start), tags: statsd_tags << "status:success")
       @logger.print_summary(:success)
-    rescue DeploymentTimeoutError => error
-      @logger.summary.add_action(error.message)
+    rescue DeploymentTimeoutError
       @logger.print_summary(:timed_out)
       ::StatsD.measure('all_resources.duration', StatsD.duration(start), tags: statsd_tags << "status:timeout")
       raise
     rescue FatalDeploymentError => error
-      @logger.summary.add_action(error.message) if error.message.present?
+      @logger.summary.add_action(error.message) if error.message != error.class.to_s
```

Comment:
`error.message.present?` is always true because it defaults to the class name

This condition is a bit of a smell... we should probably make all error sources responsible for adding action snippets if appropriate. However, I tried briefly and it's not a trivial change, and I don't think it should block us releasing the timeout stuff.